experimental/traffic-portal/src/app/core/statuses/status-details/status-details.component.spec.ts

```diff
@@ -0,0 +1,97 @@
+ */
+import { HttpClientTestingModule } from "@angular/common/http/testing";
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { ReactiveFormsModule } from "@angular/forms";
+import { MatButtonModule } from "@angular/material/button";
+import { MatCardModule } from "@angular/material/card";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatGridListModule } from "@angular/material/grid-list";
+import { MatInputModule } from "@angular/material/input";
+import { BrowserDynamicTestingModule } from "@angular/platform-browser-dynamic/testing";
+import { Router } from "@angular/router";
+import { RouterTestingModule } from "@angular/router/testing";
+
+import { ServerService } from "src/app/api";
+import { DecisionDialogComponent } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import { SharedModule } from "src/app/shared/shared.module";
+
+import { StatusDetailsComponent } from "./status-details.component";
+
+const status = { description: "test", id: 1,lastUpdated: new Date("02/02/2023"), name: "test"};
+
+describe("StatusDetailsComponent", () => {
+	let component: StatusDetailsComponent;
+	let fixture: ComponentFixture<StatusDetailsComponent>;
+	let router: Router;
+
+	beforeEach(async () => {
+		await TestBed.configureTestingModule({
+			declarations: [StatusDetailsComponent, DecisionDialogComponent],
+			imports: [
+				HttpClientTestingModule,
+				RouterTestingModule.withRoutes([
+					{ component: StatusDetailsComponent, path: "core/statuses/:id" }
+				]),
+				ReactiveFormsModule,
+				MatFormFieldModule,
+				MatInputModule,
+				MatGridListModule,
+				MatCardModule,
+				MatButtonModule,
+				SharedModule
+			],
+			providers: [ServerService]
+		})
+			.compileComponents();
+		TestBed.overrideModule(BrowserDynamicTestingModule, {
+			set: {
+				entryComponents: [DecisionDialogComponent]
+			}
+		});
+		router = TestBed.inject(Router);
+		fixture = TestBed.createComponent(StatusDetailsComponent);
+		component = fixture.componentInstance;
+		fixture.detectChanges();
+	});
+
+	it("should create", () => {
+		expect(component).toBeTruthy();
+	});
+
+	it("submits a update status request", (() => {
+		const service = TestBed.inject(ServerService);
+		component.statusDetailsForm.setValue(status);
+		spyOn(service, "updateStatusDetail").and.returnValue(Promise.resolve(status));
+		component.updateStatus();
+
+		service.updateStatus(component.statusDetailsForm.value, "1").then((result) => {
+			expect(result).toEqual(status);
+		});
+	}));
+
+	it("submits a status creation request", (() => {
+		const service = TestBed.inject(ServerService);
+		component.statusDetailsForm.setValue(status);
+
+		spyOn(service, "createStatus").and.returnValue(Promise.resolve(status));
+		component.createStatus();
+
+		service.createStatus(component.statusDetailsForm.value).then((result) => {
+			expect(result).toEqual(status);
+			router.navigate(["/core/statuses/1"]);
+		});
+	}));
```

Comment:
In these tests, you describe it as "submitting a ... request", but what you do is make a spy, then call the component method that should call the spy, but you don't verify that behavior, instead you call the spy directly and verify that it gives you what you told it to give you. We don't need to test Jasmine utilities, these tests should verify some behavior of the component.